File: java/search/array/BinarySearch.java
package search.array;

public class BinarySearch {
    public int search(int[] nums, int target) {
        int start = 0;
        int end = nums.length - 1;
        while (start <= end) {
            int middle = start + ((end - start) >> 1);
            if (target == nums[middle]) {
                return middle;
            }

            if (target < nums[middle]) {
                end = middle - 1;
            } else {
                start = middle + 1;
            }
        }
        return -1;
    }
}

File: py/search/array/BinarySearch.py
#!/usr/bin/env python
# -*- coding:gbk -*-

from typing import List

class BinarySearch:
    def search(self, nums: List[int], target: int) -> int:
        start, end = 0, len(nums) - 1
        while start <= end:
            middle = start + ((end - start) >> 1)
            if target == nums[middle]:
                return middle

            if target < nums[middle]:
                end = middle - 1
            else:
                start = middle + 1
        return -1
